saleor/warehouse/availability.py

```diff
@@ -77,11 +77,12 @@ def check_stock_quantity_bulk(
         )
 
         if not stocks:
-            insufficient_stocks.append(
-                InsufficientStockData(
-                    variant=variant, available_quantity=available_quantity
+            if quantity != 0:
+                insufficient_stocks.append(
+                    InsufficientStockData(
+                        variant=variant, available_quantity=available_quantity
+                    )
                 )
-            )
         elif variant.track_inventory:
             if quantity > available_quantity:
```

Comment:
And what about this case? Won't trigger the same exception? 
Assuming:
quantity == 0
available_quantity == 0
variant.track_inventory == True

Reply:
Maybe we run all checks only when quantity is positive:
```
if quantity > 0:
    if not stocks:
        # raise error
    elif variant.track_inventory and quantity > available_quantity:
       # raise error
```